File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        auto maxel=max_element(nums.begin(),nums.end());
        int a=1,c=0;

        for (int num:nums)
        {
            if (num==*maxel)
            {
                c+=1;
            if (c>a)
                a=c;
            }
            else 
                c=0;
        }
        return a;
        
    }
};

File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_and=ans=maxlen=0
        for num in nums:
            if max_and<num:
                max_and=num
                ans=maxlen=0
            if max_and==num:
                maxlen+=1
            else :
                maxlen=0
            ans=max(ans,maxlen)
        return ans
            